CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -290,6 +307,28 @@ namespace CefSharp
             }			
         }
 
+        void ClientAdapter::OnResourceRedirect(CefRefPtr<CefBrowser> browser, CefRefPtr<CefFrame> frame, const CefString& oldUrl, CefString& newUrl)
+        {
+            IRequestHandler^ handler = _browserControl->RequestHandler;
+            if (handler != nullptr)
+            {
+                auto managedNewUrl = StringUtils::ToClr(newUrl);
+                handler->OnResourceRedirect(_browserControl, frame->IsMain(), StringUtils::ToClr(oldUrl), managedNewUrl);
+
+                //TODO: Determine if there is a better way to pass the reference in rather than reassigning the string
+                newUrl = StringUtils::ToNative(managedNewUrl);
```

Comment:
I wasn't overly optimistic, just though I'd ask incase you'd come across something. 

I'd like to avoid doing the `newUrl = StringUtils::ToNative(managedNewUrl);` unless the url actually changed. I was thinking maybe return a bool and only assign if the user returns true.